src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -374,13 +374,13 @@ impl Step for Llvm {
             cfg.define("LLVM_PROFDATA_FILE", path);
         }
 
-        // Disable zstd to avoid a dependency on libzstd.so.
-        cfg.define("LLVM_ENABLE_ZSTD", "OFF");
-
+        // Libraries for ELF compression.
         if !target.is_windows() {
             cfg.define("LLVM_ENABLE_ZLIB", "ON");
+            cfg.define("LLVM_ENABLE_ZSTD", "ON");
         } else {
             cfg.define("LLVM_ENABLE_ZLIB", "OFF");
+            cfg.define("LLVM_ENABLE_ZSTD", "OFF");
```

Comment:
How about making this configurable in config.toml (off by default) instead of setting it on constantly?

Reply:
The point of this PR to enable it by default. If it's not enabled by default it's easier to disable the self-contained linker than to build rustc, including LLVM, from scratch.

Reply:
I do not understand why we would make something configurable without a specific reason to do so.

Reply:
Doesn't this require an additional dependency to be installed? Also, according to the LLVM docs when it's not set LLVM build enables zstd only if it exists on the system. Could that be an option here?

Fwiw using "FORCE_ON" makes the build raise a hard error when zstd can't be found on the system.

ref: https://llvm.org/docs/CMake.html

Reply:
> Doesn't this require an additional dependency to be installed?

Raising the requirements to build our LLVM recipe seems fine given that we support using an external LLVM. Thus we can't assume that the external LLVM has zlib **or** zstd support (as both are optional) and must query it at runtime, and we support building an LLVM with arbitrary configuration already: just bring your own.

Reply:
> Doesn't this require an additional dependency to be installed?

As the patch stands today, it requires libzstd.so to be available at runtime if libzstd was available at build time. It doesn't actually require libzstd to be available at build time (because it uses ON instead of FORCE_ON).

There's a matrix of possible outcomes here.

|| ON | FORCE_ON|
|-|-----|-------------|
| No LLVM_USE_STATIC_ZSTD | Not required at build time, required at runtime if present at build time, silently disabled if not present at build time | Required at build time and runtime|
| LLVM_USE_STATIC_ZSTD | Not required at build time or runtime, silently disabled if not present at build time | Required at  build time, not at runtime|

Where we're currently in the top left cell.